modules/scheduler/gke-cluster/main.tf

```diff
@@ -75,7 +75,7 @@ resource "google_container_cluster" "gke_cluster" {
   }
 
   private_ipv6_google_access = var.enable_private_ipv6_google_access ? "PRIVATE_IPV6_GOOGLE_ACCESS_TO_GOOGLE" : null
-
+  default_max_pods_per_node  = var.max_pods_per_node
```

Comment:
Setting this parameter needs IP Alias enabled [based on the documents](https://registry.terraform.io/providers/hashicorp/google/latest/docs/resources/container_cluster#default_max_pods_per_node).
I would suggest to have a `precondition` to check IP Alias is enabled ([networking_mode](https://registry.terraform.io/providers/hashicorp/google/latest/docs/resources/container_cluster#networking_mode)=VPC_NATIVE) when this parameter is set.

Reply:
The same condition is needed in gke-node-pool module
A [container_cluster data-sources](https://registry.terraform.io/providers/hashicorp/google/latest/docs/data-sources/container_cluster) can provide the required info.